pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -82,7 +82,8 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
 	logging.FromContext(ctx).Debugf("Using AWS region %s", *sess.Config.Region)
 	ec2api := ec2.New(sess)
 	subnetProvider := NewSubnetProvider(ec2api)
-	instanceTypeProvider := NewInstanceTypeProvider(ec2api, subnetProvider)
+	pricingProvider := NewPricingProvider(ctx, NewPricingAPI(sess, *sess.Config.Region), ec2api, *sess.Config.Region)
```

Comment:
There's a lot of unnecessary duplication in this. Seems like we need the sess and ec2api.
```suggestion
	pricingProvider := NewPricingProvider(ctx, sess, ec2api)
```

Reply:
Oh -- you're using the NewPricingProvider function in tests, rather than building out the struct. Makes sense to me.